sdks/python/apache_beam/io/gcp/bigtable_io_write.py

```diff
@@ -0,0 +1,166 @@
+except ImportError:
+  pass
+
+
+class WriteToBigtable(beam.DoFn):
+  """ Creates the connector can call and add_row to the batcher using each
+  row in beam pipe line
+
+  :type beam_options: class:`~bigtable_configuration.BigtableConfiguration`
+  :param beam_options: class `~bigtable_configuration.BigtableConfiguration`
+  """
+
+  def __init__(self, beam_options):
+    super(self.__class__, self).__init__(beam_options)
+    self.beam_options = beam_options
+    self.client = None
+    self.instance = None
+    self.table = None
+    self.batcher = None
+    self._app_profile_id = self.beam_options.app_profile_id
+    self.flush_count = self.beam_options.flush_count
+    self.max_row_bytes = self.beam_options.max_row_bytes
+    self.written = Metrics.counter(self.__class__, 'Written Row')
+
+  def start_bundle(self):
+    if self.beam_options.credentials is None:
+      self.client = Client(project=self.beam_options.project_id,
+                           admin=True)
+    else:
+      self.client = Client(project=self.beam_options.project_id,
+                           credentials=self.beam_options.credentials,
+                           admin=True)
+    self.instance = self.client.instance(self.beam_options.instance_id)
+    self.table = self.instance.table(self.beam_options.table_id,
+                                     self._app_profile_id)
+
+    self.batcher = MutationsBatcher(self.table, flush_count=self.flush_count,
+                                    max_row_bytes=self.max_row_bytes)
+    self.written = Metrics.counter(self.__class__, 'Written Row')
+
+  def process(self, row):
+    self.written.inc()
+    self.batcher.mutate(row)
+
+  def finish_bundle(self):
+    return self.batcher.flush()
+
+  def display_data(self):
+    return {'projectId': DisplayDataItem(self.beam_options.project_id,
+                                         label='Bigtable Project Id'),
+            'instanceId': DisplayDataItem(self.beam_options.instance_id,
+                                          label='Bigtable Instance Id'),
+            'tableId': DisplayDataItem(self.beam_options.table_id,
+                                       label='Bigtable Table Id'),
+            'bigtableOptions': DisplayDataItem(str(self.beam_options),
+                                               label='Bigtable Options',
+                                               key='bigtableOptions'),
+           }
+
+
+class BigtableConfiguration(object):
+  """ Bigtable configuration variables.
+
+  :type project_id: :class:`str` or :func:`unicode <unicode>`
+  :param project_id: (Optional) The ID of the project which owns the
+    instances, tables and data. If not provided, will
+    attempt to determine from the environment.
+
+  :type instance_id: str
+  :param instance_id: The ID of the instance.
+
+  :type table_id: str
+  :param table_id: The ID of the table.
+  """
+
+  def __init__(self, project_id, instance_id, table_id):
```

Comment:
Please add `app_profile` to `BigtableConfiguration`.

Reply:
I resolve this problem.